activesupport/lib/active_support/cache/redis_cache_store.rb

```diff
@@ -363,7 +363,15 @@ def write_multi_entries(entries, expires_in: nil, **options)
 
         # Truncate keys that exceed 1kB.
         def normalize_key(key, options)
-          truncate_key super
+          truncate_key encode_key(super)
+        end
+
+        def encode_key(key)
+          if key.encoding == Encoding::BINARY
+            key
+          else
+            key.dup.force_encoding(Encoding::BINARY)
+          end
```

Comment:
Perhaps we could get away with just `key.b` to dup & strip encoding? (Binary keys should be uncommon, given default source encoding, so the no-op may not save us much if anything.)

Reply:
Didn’t know about `String#b`. Switched.